.github/workflows/master.yml

```
@@ -31,7 +31,7 @@ jobs:
 
       - uses: actions/setup-go@v5
         with:
-          go-version: 1.21
+          go-version-file: "go.mod"
 
       - name: Unit tests
         run: go test ./...
```

What about adding a lint action?
```suggestion
        run: go test ./...

  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-go@v5
        with:
          go-version-file: 'go.mod'

      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3.7.0
        with:
          version: latest
          args: -v --config ./.golangci.yml

```

---

We can but I would suggest doing it in a different PR as it is independent from this change